### PR TITLE
[MI-54922] Replaced the "text/template" package with "html/template"

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
Replaced the "text/template" package with "html/template"

### CODE REVIEW

**Review by mickmister — APPROVED** [review]

**Review by jupenur — CHANGES_REQUESTED** [review]
I don't think this will work since there's a mixture of HTML and non-HTML templates that we use. Only the ones that are actually HTML should use `html/template`.

**Comment by raghavaggarwal2308** [comment]
@jupenur I have tested it for "application/json" and "markdown" templates, and everything seems to work fine. Also, in the documentation of [html/template](https://pkg.go.dev/html/template), it's mentioned that this package wraps "text/template". Please let me know your opinions on this.
```
This package wraps [text/template](https://pkg.go.dev/text/template) so you can share its template API to parse and execute HTML templates safely.
```

**Comment by mickmister** [comment]
@raghavaggarwal2308 In general, we should only use the `html/template` library when we are specifically generating HTML for the response. It has different behavior than `text/template` in that it sanitizes the input in ways that we may otherwise want to preserve when the output is not intended to be HTML

**Comment by raghavaggarwal2308** [comment]
@mickmister @jupenur Updated logic to use 'html/template' for HTML templates only.

File: server/command_test.go [hunk 1]
```diff
@@ -374,9 +374,10 @@ func TestPlugin_ExecuteCommand_Installation(t *testing.T) {
 			p.SetAPI(api)
 			p.client = pluginapi.NewClient(p.API, p.Driver)
 			_, filename, _, _ := runtime.Caller(0)
-			templates, err := p.loadTemplates(filepath.Dir(filename) + "/../assets/templates")
+			htmlTemplates, textTemplates, err := p.loadTemplates(filepath.Dir(filename) + "/../assets/templates")
 			require.NoError(t, err)
-			p.templates = templates
+			p.htmlTemplates = htmlTemplates
+			p.textTemplates = textTemplates
```
Comment (ayusht2810) [inline server/command_test.go:380]:
```suggestion
			p.htmlTemplate = htmlTemplate
			p.textTemplate = textTemplate
```

File: server/http.go [hunk 2]
```diff
@@ -192,30 +212,61 @@ func respondJSON(w http.ResponseWriter, obj interface{}) (int, error) {
...
+
+		if err := t.Execute(w, values); err != nil {
+			return http.StatusInternalServerError,
+				errors.WithMessage(err, "failed to write response")
+		}
+
+		return status, nil
+	}
+
+	t := p.textTemplates[key]
```
Comment (ayusht2810) [inline server/http.go:259]: similar snippet is present above as well. Should we create a common function for both?

File: server/command_test.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/command_test.go:380]: @ayusht2810 These two fields are a map of templates, not a single template. So, I think what we have is fine. Let me know your opinions.

**Review by mickmister — APPROVED** [review]
LGTM

File: server/http.go (hunk 2 above)
Comment (raghavaggarwal2308) [reply server/http.go]: @ayusht2810 Created a common function

**Review by ayusht2810 — APPROVED** [review]

**Review by jupenur — APPROVED** [review]

File: server/http.go [hunk 3]
```diff
@@ -145,30 +149,46 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
...
+				p.errorf("OnActivate: failed to parse template %s: %v", path, err)
+				return nil
+			}
+
+			key = path[len(dir):]
+			htmlTemplates[key] = template
+		} else {
+			template, err := textTemplate.ParseFiles(path)
+			if err != nil {
+				p.errorf("OnActivate: failed to parse template %s: %v", path, err)
```
Comment (avas27JTG) [inline server/http.go:177]:
```suggestion
				p.errorf("OnActivate: failed to parse the template %s: %v", path, err)
```

**Review by avas27JTG — APPROVED** [review]

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested and Approved
Working fine for both cloud and server, LGTM

**Pull request merged**